src/dart/ast.rs

```diff
@@ -212,34 +301,53 @@ impl FnName {
     }
 }
 
+/// Overloaded operators.
 #[derive(Copy, Clone, Debug)]
 pub enum OverloadedOp {
+    /// Bitwise negation. (`operator~`)
     BitNot,
+    /// Indexing. (`operator[]`)
     Index,
+    /// Indexed assignment. (`operator[]=`)
     IndexAssign,
+    /// Boolean binary operator.
     Bool(BoolBinOp),
+    /// Value binary operator.
     Value(ValueBinOp),
 }
 
+/// An element of constructor initializers list.
 #[derive(Debug)]
 pub enum ConstructorInitializer {
+    /// Redirects to the superclass' constructor.
     Super(Option<Symbol>, Args),
+    /// Redirects to the current class' constructor.
     This(Option<Symbol>, Args),
+    /// Precondition to constructing the object.
     Assert(Args),
+    /// Initializes a specific field (`Person():name('Name') {...}`).
     Field(bool, Symbol, Node<Expr>),
 }
 
+/// Type parameter definition.
 #[derive(Debug)]
 pub struct TypeParameter {
+    /// Metadata of the parameter.
     pub meta: Meta,
+    /// The name of the parameter.
     pub name: Symbol,
+    /// Limits the type of the parameter (`<T extends V>`).
     pub extends: Option<Node<Qualified>>,
 }
 
+/// Optionally fully-qualified path.
 #[derive(Debug)]
 pub struct Qualified {
+    /// The object to look up the name in.
     pub prefix: Option<Node<Qualified>>,
+    /// The name of the object.
     pub name: Symbol,
+    /// The type parameters of the object (`List<int> Map<K,V>`).
```

Comment:
These are two examples, so they should be in separate code snippets, with a comma between them.